### PR TITLE
Update Dataset Formatting guide content

### PR DESCRIPTION
**Pull request by sandalns**
### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

File: docs/dataset/formatting-your-datasets.md [hunk 1]
```diff
@@ -90,13 +98,51 @@ These annotations are visible on both the Gallery view for groups of datapoints
...
+from kolena.annotation import BoundingBox
+bboxes = [
+    BoundingBox(top_left=(270.77, 44.59), bottom_right=(621.61, 254.18)),
+    BoundingBox(top_left=(538.03, 8.86), bottom_right=(636.85, 101.93)),
+    BoundingBox(top_left=(313.02, 12.01), bottom_right=(553.98, 99.84)),
+]
+```
+This would be represented within a csv as show below. Note this will be a single line,
+but is shown here as multiple lines for formatting.
+```
```
Comment (sandalns) [inline docs/dataset/formatting-your-datasets.md:141]: Thinking about the best way to show this is a single row. Needed to split for both the markdown length rules and the fact that a single line this long does not have a good experience for the reader.

File: docs/dataset/formatting-your-datasets.md [hunk 2]
```diff
@@ -23,26 +23,34 @@ dataset with the following columns:
...
+The table below shows the supported file formats for the supported data types:
+
 | Data Type   | Supported file formats                                                                |
 |-------------|---------------------------------------------------------------------------------------|
 | Image       | `jpg`, `jpeg`, `png`, `gif`, `bmp` and other web browser supported image types.       |
 | Audio       | `flac`, `mp3`, `wav`, `acc`, `ogg`, `ra` and other web browser supported audio types. |
 | Video       | `mov`, `mp4`, `mpeg`, `avi` and other web browser supported video types.              |
 | Document    | `txt` and `pdf` files.                                                                |
-| Point Cloud | `pcd` files.                                                                          |
+| Point Cloud | `pcd` files.
```
Comment (munkyshi) [inline docs/dataset/formatting-your-datasets.md]: Did this lose the end pipe?

File: docs/dataset/formatting-your-datasets.md [hunk 3]
```diff
@@ -23,26 +23,34 @@ dataset with the following columns:
...
+When uploading a dataset to Kolena, it is important to be able to differentiate between each datapoint. This is
+accopmplished by configuring an `id_field` - an unique identifier for a datapoint. You can select any field that is
+unique across your data, for example a `uuid` or `article_id` for relevant data.
+
+Kolena handles the following `id_fields` as a special case:
+
+| Field Name  | Description                                                                                                                              |
+|-------------|------------------------------------------------------------------------------------------------------------------------------------------|
+| `locator`     | Url path to a file to be displayed, either a [cloud storage](../connecting-cloud-storage/index.md) url or a http url that serves a file. |
+| `text`        | Raw text input for text based models.                                                                                                    |
```
Comment (consoull) [inline docs/dataset/formatting-your-datasets.md]: These aren't necessarily `id_fields`. Can this be rewritten to something like "Kolena will look for the following fields when displaying datapoints"

Comment (consoull) [reply docs/dataset/formatting-your-datasets.md]: nit: can we keep tables aligned

File: docs/dataset/formatting-your-datasets.md [hunk 4]
```diff
@@ -23,26 +23,34 @@ dataset with the following columns:
...
 such as `.jpg` or `.png`, whereas locators for audio data should be in forms like `.mp3` or `.wav`.
 
-| Data Type   | Supported file formats                                                                |
-|-------------|---------------------------------------------------------------------------------------|
-| Image       | `jpg`, `jpeg`, `png`, `gif`, `bmp` and other web browser supported image types.       |
-| Audio       | `flac`, `mp3`, `wav`, `acc`, `ogg`, `ra` and other web browser supported audio types. |
-| Video       | `mov`, `mp4`, `mpeg`, `avi` and other web browser supported video types.              |
-| Document    | `txt` and `pdf` files.                                                                |
-| Point Cloud | `pcd` files.                                                                          |
+The table below shows the supported file formats for the supported data types:
```
Comment (consoull) [inline docs/dataset/formatting-your-datasets.md]: nit: can we drop one of the `supported` here?
I feel this can be replaces with a title for the table eg. `### Locator Support Matrix`

File: docs/dataset/formatting-your-datasets.md [hunk 5]
```diff
@@ -23,26 +23,34 @@ dataset with the following columns:
...
+| Image          | `jpg`, `jpeg`, `png`, `gif`, `bmp` and other web browser supported image types.       |
+| Audio          | `flac`, `mp3`, `wav`, `acc`, `ogg`, `ra` and other web browser supported audio types. |
+| Video          | `mov`, `mp4`, `mpeg`, `avi` and other web browser supported video types.              |
+| Document       | `txt` and `pdf` files.                                                                |
+| Point Cloud    | `pcd` files.                                                                          |
+
+By default the `locator` or  `text` fields are used if present in your dataset, and other fields
+can be specified when importing via the Web App from the [:kolena-dataset-16: Datasets](https://app.kolena.io/redirect/datasets)
+page, or the SDK by using [`upload_dataset`](../reference/dataset/index.md#kolena.dataset.dataset.upload_dataset)
+function.
```
Comment (consoull) [inline docs/dataset/formatting-your-datasets.md]: Not clear this is talking about inferring `id_fields`. Can we be more explicit here eg "Kolena will attempt to infer common id fields (eg. `locator`, `text`) based on what is present in the dataset during import. This can be overriden by explicitly declaring id fields."

File: docs/dataset/formatting-your-datasets.md [hunk 6]
```diff
@@ -90,14 +98,63 @@ These annotations are visible on both the Gallery view for groups of datapoints
...
+
+A single bounding box would be serialized as the following JSON string within a CSV:
+
+```
+{""top_left"": [270.77, 44.59], ""bottom_right"": [621.61, 254.18], ""width"": 350.84, ""height"": 209.59,
+ ""area"": 73532.5556, ""aspect_ratio"": 1.67, ""data_type"": ""ANNOTATION/BOUNDING_BOX""},
+```
+
+The above example has multiple objects within a single image, in order to ingest this you would need a single field
+containing the list of bounding boxes.
```
Comment (consoull) [inline docs/dataset/formatting-your-datasets.md]: I know the answer but this might make someone question why they can't use multiple fields for each bounding box.
Can we shorten this to  "The above example has multiple objects within a single image. This can be represented by a list of bounding boxes."

File: docs/dataset/formatting-your-datasets.md (hunk 1 above)
Comment (consoull) [reply docs/dataset/formatting-your-datasets.md:141]: What's the benefit in dumping this blob in the documentation? Can't we just say the Kolena annotation class is serialized to json or something

File: docs/dataset/formatting-your-datasets.md [hunk 7]
```diff
@@ -90,14 +98,63 @@ These annotations are visible on both the Gallery view for groups of datapoints
...
+```
+
 When uploading `.csv` files for datasets that contain annotations, assets or nested values in a column use the
 [`dataframe_to_csv()`](../reference/io.md#kolena.io.dataframe_to_csv) function provided by Kolena to save a `.csv` file
-instead of [`pandas.to_csv()`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_csv.html).
-
-In order to add data like a list of `BoundingBox` objects to your dataset via the sdk all you need to do is have
-field with a list of objects in your dataframe.
+instead of [`pandas.to_csv()`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_csv.html). The above
+serializations were created using `dataframe_to_csv()`.
```
Comment (consoull) [inline docs/dataset/formatting-your-datasets.md]: Should probably have a brief blurb that `pandas.to_csv` won't properly serialize Kolena annotation objects. Right now we are just saying don't use it without an explanation.

nit: I think we can leave out: "The above serializations were created using `dataframe_to_csv()`."

File: docs/dataset/formatting-your-datasets.md [hunk 8]
```diff
@@ -129,21 +186,28 @@ df_merged = df_metadata.merge(df_boxes, on="locator")
...
 |------------------------|-----------------------------------------------------|
-| matched_inference      | Inferences that were matched to a ground truth.     |
-| unmatched_inference    | Inferences that were not matched to a ground truth. |
-| unmatched_ground_truth | Inferences that were not matched to a ground truth. |
+| `matched_inference`      | Inferences that were matched to a ground truth.     |
+| `unmatched_inference`    | Inferences that were not matched to a ground truth. |
+| `unmatched_ground_truth` | Inferences that were not matched to a ground truth. |
 
 These columns are used to determine `True Postitives`, `False Positives`, and `False Negatives`.
+These results can be formated for upload with a similar process as above. This is done by adding the relavent list of
```
Comment (consoull) [inline docs/dataset/formatting-your-datasets.md]: `formated` -> `formatted`, `relavent` -> `relevant`. Can we run a spell checker over the changes?

Comment (sandalns) [reply docs/dataset/formatting-your-datasets.md]: Ran a spell checker over the changes and updated. In a follow up going to look at adding a spell check to the pre-commit for docs.

File: docs/dataset/formatting-your-datasets.md (hunk 3 above)
Comment (sandalns) [reply docs/dataset/formatting-your-datasets.md]: updated

File: docs/dataset/formatting-your-datasets.md (hunk 5 above)
Comment (sandalns) [reply docs/dataset/formatting-your-datasets.md]: updated

File: docs/dataset/formatting-your-datasets.md (hunk 6 above)
Comment (sandalns) [reply docs/dataset/formatting-your-datasets.md]: Updated this

File: docs/dataset/formatting-your-datasets.md (hunk 7 above)
Comment (sandalns) [reply docs/dataset/formatting-your-datasets.md]: Updated

File: docs/dataset/formatting-your-datasets.md (hunk 4 above)
Comment (sandalns) [reply docs/dataset/formatting-your-datasets.md]: Added a title for the table, good idea.

File: docs/dataset/formatting-your-datasets.md (hunk 1 above)
Comment (sandalns) [reply docs/dataset/formatting-your-datasets.md:141]: The idea was to show the user how to go from the raw data to a .csv that can be uploaded. The purpose of adding the blob was to help users understand how the json itself looks, and to have an idea of how the data looks at each step.

File: docs/dataset/formatting-your-datasets.md
Comment (munkyshi) [inline docs/dataset/formatting-your-datasets.md:1]: Line 15:
> Conceptually a datapoint is a set of inputs that you would want to test on your models.

should have a comma

> Conceptually, a datapoint is a set of inputs that you would want to test on your models.

File: docs/dataset/formatting-your-datasets.md [hunk 9]
```diff
@@ -23,26 +23,34 @@ dataset with the following columns:
...
+| Audio          | `flac`, `mp3`, `wav`, `acc`, `ogg`, `ra` and other web browser supported audio types. |
+| Video          | `mov`, `mp4`, `mpeg`, `avi` and other web browser supported video types.              |
+| Document       | `txt` and `pdf` files.                                                                |
+| Point Cloud    | `pcd` files.                                                                          |
 
-For text-based models the `text` field contains the raw text input for the models.
+Kolena will attempt to infer common id fields (eg. locator, text) based on what is present in the dataset during import.
+This can be overriden by explicitly declaring id fields when importing via the Web App from the [:kolena-dataset-16: Datasets](https://app.kolena.io/redirect/datasets)
+page, or the SDK by using the [`upload_dataset`](../reference/dataset/index.md#kolena.dataset.dataset.upload_dataset)
+function.
```
Comment (munkyshi) [inline docs/dataset/formatting-your-datasets.md]: I feel like this should be above line 30 -- it's a bit of a jump to go from talking about id fields, to details for locators, to back to id fields.

File: docs/dataset/formatting-your-datasets.md [hunk 10]
```diff
@@ -23,26 +23,34 @@ dataset with the following columns:
...
+| Data Type      | Supported file formats                                                                |
+|----------------|---------------------------------------------------------------------------------------|
+| Image          | `jpg`, `jpeg`, `png`, `gif`, `bmp` and other web browser supported image types.       |
+| Audio          | `flac`, `mp3`, `wav`, `acc`, `ogg`, `ra` and other web browser supported audio types. |
+| Video          | `mov`, `mp4`, `mpeg`, `avi` and other web browser supported video types.              |
+| Document       | `txt` and `pdf` files.                                                                |
+| Point Cloud    | `pcd` files.                                                                          |
 
-For text-based models the `text` field contains the raw text input for the models.
+Kolena will attempt to infer common id fields (eg. locator, text) based on what is present in the dataset during import.
```
Comment (munkyshi) [inline docs/dataset/formatting-your-datasets.md]:
```suggestion
Kolena will attempt to infer common `id_field`s (eg. `locator`, `text`) based on what is present in the dataset during import.
```

File: docs/dataset/formatting-your-datasets.md [hunk 11]
```diff
@@ -60,7 +68,7 @@ The second experience is the Tabular view, used when your data is a set of colum
 An example of this is the [:kolena-widget-16: Rain Forcast ↗](https://github.com/kolenaIO/kolena/tree/trunk/examples/dataset/rain_forecast)
 dataset.
 
-In order to use the Gallery view you just need to have the `locator` or `text` fields specfied in the dataset.
+In order to use the Gallery view you just need to have the `locator` or `text` fields specified in the dataset.
```
Comment (munkyshi) [inline docs/dataset/formatting-your-datasets.md]:
```suggestion
In order to use the Gallery view you will need to have the `locator` or `text` fields specified in the dataset.
```

File: docs/dataset/formatting-your-datasets.md [hunk 12]
```diff
@@ -23,26 +23,34 @@ dataset with the following columns:
 From this you can see that image `horse0000.png` has the ground_truth classification of `horse`,
 and has brightness and contrast data.
 
-An `id_field` is required in order to differentiate between datapoints.
-By default the `locator` or  `text` fields are used if present in your dataset, and other fields
-can be specified when importing via the Web App from the [:kolena-dataset-16: Datasets](https://app.kolena.io/redirect/datasets)
-page, or the SDK by using [`upload_dataset`](../reference/dataset/index.md#kolena.dataset.dataset.upload_dataset)
-function.
+When uploading a dataset to Kolena, it is important to be able to differentiate between each datapoint. This is
+accomplished by configuring an `id_field` - an unique identifier for a datapoint. You can select any field that is
```
Comment (consoull) [inline docs/dataset/formatting-your-datasets.md]: `an unique` -> `a unique`

File: docs/dataset/formatting-your-datasets.md [hunk 13]
```diff
@@ -23,26 +23,34 @@ dataset with the following columns:
...
+unique across your data, for example a `uuid` or `article_id` for relevant data.
```
Comment (consoull) [inline docs/dataset/formatting-your-datasets.md]: Should we say "...or generate a `uuid` if no unique identifiers exist for your dataset"? Seems odd to group a generated id in with a natural id like locator/article_id

File: docs/dataset/formatting-your-datasets.md (hunk 11 above)
Comment (consoull) [inline docs/dataset/formatting-your-datasets.md]: Should we give a name to these display fields so we can reference the above table with `locator` and `text` here?

File: docs/dataset/formatting-your-datasets.md [hunk 14]
```diff
@@ -129,21 +187,28 @@ df_merged = df_metadata.merge(df_boxes, on="locator")
...
-| unmatched_inference    | Inferences that were not matched to a ground truth. |
-| unmatched_ground_truth | Inferences that were not matched to a ground truth. |
+| `matched_inference`      | Inferences that were matched to a ground truth.     |
+| `unmatched_inference`    | Inferences that were not matched to a ground truth. |
+| `unmatched_ground_truth` | Inferences that were not matched to a ground truth. |
 
 These columns are used to determine `True Postitives`, `False Positives`, and `False Negatives`.
+These results can be formatted for upload with a similar process as above. This is done by adding the relevant list of
+bounding boxes to the `matched_inference`, `unmatched_inference`, and `unmatched_ground_truth` columns for each image.
+The results.csv created can be uploaded by opening the corresponding dataset from the [:kolena-dataset-16: Datasets](https://app.kolena.io/redirect/datasets)
```
Comment (consoull) [inline docs/dataset/formatting-your-datasets.md]: csv files have been consistently quoted else where eg. results.csv -> `results.csv`

**Review by consoull — APPROVED** [review]

File: docs/dataset/formatting-your-datasets.md (hunk 12 above)
Comment (sandalns) [reply docs/dataset/formatting-your-datasets.md]: done

File: docs/dataset/formatting-your-datasets.md (hunk 14 above)
Comment (sandalns) [reply docs/dataset/formatting-your-datasets.md]: done

File: docs/dataset/formatting-your-datasets.md (hunk 13 above)
Comment (sandalns) [reply docs/dataset/formatting-your-datasets.md]: done

File: docs/dataset/formatting-your-datasets.md (hunk 11 above)
Comment (sandalns) [reply docs/dataset/formatting-your-datasets.md]: Will think about this, the best name I came up with was `Kolena default display field ` or `Kolena default id fields`

Comment (sandalns) [reply docs/dataset/formatting-your-datasets.md]: updated

File: docs/dataset/formatting-your-datasets.md (hunk 10 above)
Comment (sandalns) [reply docs/dataset/formatting-your-datasets.md]: updated

File: docs/dataset/formatting-your-datasets.md (hunk 9 above)
Comment (sandalns) [reply docs/dataset/formatting-your-datasets.md]: moved to above line 30

File: docs/dataset/formatting-your-datasets.md
Comment (sandalns) [reply docs/dataset/formatting-your-datasets.md:1]: updated

File: docs/dataset/formatting-your-datasets.md (hunk 2 above)
Comment (sandalns) [reply docs/dataset/formatting-your-datasets.md]: this was added back last week

**Review by munkyshi — APPROVED** [review]
